app/models/piece.rb

```diff
@@ -1,53 +1,93 @@
     # If not error is raised.
-    if (x_coordinate - x).abs == (y_coordinate - y).abs or x == x_coordinate or y == y_coordinate
-      # Check each item in given array for an obstruction using contains_piece? method.
+    if (x_coordinate - x).abs == (y_coordinate - y).abs || x == x_coordinate || y == y_coordinate
+    # Check each item in given array for an obstruction using contains_piece? method.
       pathway_array(x, y).any? {|h, v| game.contains_piece?(h, v) }
     else
-      raise ArgumentError, 'Invalid input. Not diagonal, horizontal, or vertical.'
+      raise ArgumentError, 'Invalid move: not diagonal, horizontal, or vertical.'
     end
   end
 
-  # Outputs an array of coordinates between Piece and destination.
-  # Vertical, horizontal and diagonal movements only.
-  # Does not include Piece or destination coordinates.
+# Outputs an array of coordinates between Piece and destination.
+# Vertical, horizontal and diagonal movements only.
+# Does not include Piece or destination coordinates.
   def pathway_array(x, y)
     pathway_spaces = []
-    # Make coordinates available locally to avoid erroneous changes.
+
+  # Make coordinates available locally to avoid erroneous changes.
     pos_x = x_coordinate
     pos_y = y_coordinate
-    # Determine the increments based on destination position relative to Piece.
+
+  # Determine the increments based on destination position relative to Piece.
     x_increment = x <=> x_coordinate
     y_increment = y <=> y_coordinate
-    # Determine if pathway is horizontal, vertical or diagonal then loop.
+
+  # Determine if pathway is horizontal, vertical or diagonal then loop.
     if (x_coordinate - x).abs == (y_coordinate - y).abs
-      # Diagonal pathway loop
+    # Diagonal pathway loop
       while (x - pos_x).abs > 0 && (y - pos_y).abs > 0
         pathway_spaces << [pos_x, pos_y]
         pos_x += x_increment
         pos_y += y_increment
       end
     elsif x == x_coordinate
-      # Vertical pathway loop
+    # Vertical pathway loop
       while (y - pos_y).abs > 0
         pathway_spaces << [pos_x, pos_y]
         pos_y += y_increment
       end
     else
       while (x - pos_x).abs > 0
-        # Horizontal pathway loop
+      # Horizontal pathway loop
         pathway_spaces << [pos_x, pos_y]
         pos_x += x_increment
       end
     end
-    # Delete first position which is Piece's position.
-    pathway_spaces.delete_at(0)
+  # Delete first position, which is Piece's position.
+    pathway_spaces.delete_at 0
     pathway_spaces
   end
-end
\ No newline at end of file
+
+  def friendly_piece_occupies_destination?(x, y)
+  # Query for friendly piece in target destination
+    friendly_piece = game.pieces.where(x_coordinate: x, y_coordinate: y, color: color).last
+  
+  # Check if a friendly piece occupies the targeted destination  
+    friendly_piece ? true : false
+  end
+
+  def enemy_piece_captured?(x, y)
```

Comment:
not sure I agree with the name. The `?` makes sense, but maybe `capture_piece?` instead?. I also feel like it shouldn't check if the occupying pieces is of the right color anyway (since when we call this we've already checked).